spec/features/user_can_cancel_correction_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.feature 'Cancelling a correction submission' do
+  around(:each) do |example|
+    ClimateControl.modify CORRECTION_RETURNS_ENABLED: 'yes' do
+      example.run
+    end
+  end
+
+  context 'for a task with a correction in progress' do
+    scenario 'it can be cancelled' do
+      visit '/'
+
+      mock_user_endpoint!
+      mock_sso_with(email: 'email@example.com')
+      mock_incomplete_tasks_endpoint!
+      click_link 'sign-in'
+
+      mock_task_with_framework_endpoint!(task_id: 'b847e0f7-027e-4b95-afa2-3490b8d05a1d')
+      click_link 'Cancel correction'
+      mock_task_endpoint!(task_id: 'b847e0f7-027e-4b95-afa2-3490b8d05a1d')
+      mock_correction_cancellation = mock_correction_cancellation_endpoint!
+      click_button 'Cancel correction'
+
+      expect(mock_correction_cancellation).to have_been_requested
```

Comment:
I'm confused as to how this works as the mock is mocking with the `mock_task_id`. Shouldn't it be mocking using the ID above for the 'correcting' task?

Reply:
So, you found our secret guilt here. The previous mock returns the 'standard' task ID, as it is hardcoded in the fixture file, which we can't change with a method attribute, as the fixtures are not dynamic in any way.

So in reality, we are switching task IDs halfway through our multi-request feature spec here. This is definitely not good.

We'd like to hear suggestions on this though. Do we start dynamically loading the fixtures in api_helper, and passing in changes to them in method attributes? Do we duplicate the methods and fixtures to create every possible scenario we need in our tests? Do we try and avoid both of the above in the slightly gnarly way we have here?

Overall, I would like to have a proper rethink about how we do the stubbing of the API as a whole. This didn't feel like the moment to do that though. [james]

Reply:
I think I'm ok with adding the extra mock methods and JSON file(s) that accurately represent the "correcting" task. I'd prefer the extra explicit methods over confusing reuse of mocks that return tasks in the incorrect states.